### PR TITLE
Docs for Cypress.Commands.overwriteQuery

### PR DESCRIPTION
**Pull request by BlueWinds**
- Closes https://github.com/cypress-io/cypress-documentation/issues/4881

Matching code PR: https://github.com/cypress-io/cypress/pull/25674

See that PR for more details.

### CODE REVIEW

File: docs/api/cypress-api/ensure.mdx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+```
+
+:::caution
+
+Many of these functions accept an optional `onFail` argument. This is a legacy
+feature used to customize the thrown error, and may be removed in a future
+release; we recommend against relying on it. If you need more control over the
+error thrown, write your own `ensure` function instead.
+
+:::
```
Comment (emilyrohrbough) [inline docs/api/cypress-api/ensure.mdx:44]: Lets just not even document this if we ever want to remove it.

File: docs/api/cypress-api/ensure.mdx [hunk 2]
```diff
@@ -0,0 +1,81 @@
+---
```
Comment (emilyrohrbough) [inline docs/api/cypress-api/ensure.mdx:1]: Can we break these docs into their own PR?

File: docs/api/cypress-api/ensure.mdx (hunk 1 above)
Comment (BlueWinds) [reply docs/api/cypress-api/ensure.mdx:44]: The reason I documented it here is because people's intellisense / TS will show them that there's an additional argument, and they'll see it if they examine the code.

Undocumented arguments are the worst, so I figure a little warning "this exists but don't use it" would be preferable.

File: docs/api/cypress-api/ensure.mdx (hunk 2 above)
Comment (BlueWinds) [reply docs/api/cypress-api/ensure.mdx:1]: https://github.com/cypress-io/cypress-documentation/pull/5054

File: docs/api/cypress-api/custom-queries.mdx [hunk 3]
```diff
@@ -55,6 +56,10 @@ Cypress.Commands.addQuery(name, callbackFn)
 Cypress.Commands.addQuery('getById', function (id) {
   return (subject) => newSubject
 })
+
+Cypress.Commands.addQuery('get', function (originalFn, ...args) {
```
Comment (flotwig) [inline docs/api/cypress-api/custom-queries.mdx:60]:
```suggestion
Cypress.Commands.overwriteQuery('get', function (originalFn, ...args) {
```

File: docs/api/cypress-api/custom-queries.mdx [hunk 4]
```diff
@@ -306,6 +311,70 @@ be executed later (which is not idempotent!). `cy.now()` lets us directly call
...
+`Cypress.Commands.addQuery` - it is a **function that returns a function.** This
+gives you access to both the outer arguments (before you call `originalFn`) and
+the inner function (the return value of `originalFn`), giving you a great deal
+of control over how the query executes.
+
+Remember that query functions rely on `this` - when you invoke `originalFn`, be
+sure to use `.call` or `.apply`.
+
+#### Adding alias support to `.contains()`
+
```
Comment (flotwig) [inline docs/api/cypress-api/custom-queries.mdx:351]: I think it'd be nice to explain this code segment briefly:

```suggestion

In this example, `cy.contains()` is extended to support querying for aliased subjects, like `cy.contains('@foo')`.

```

**Review by flotwig — APPROVED** [review]

File: docs/api/cypress-api/custom-queries.mdx (hunk 4 above)
Comment (BlueWinds) [reply docs/api/cypress-api/custom-queries.mdx:351]: Good call.

**Review by mjhenkes — APPROVED** [review]
